builder/api/src/main/java/io/helidon/builder/api/Prototype.java

```diff
@@ -450,5 +451,39 @@ public interface OptionDecorator<B, T> {
          */
         String[] value();
     }
+
+    /**
+     * Utility class used from generated code for equals and hash code of specific cases of optional.
+     * This exists because of Optional of char arrays, where equals does not work as we need.
+     */
+    public static class OptionalUtil {
```

Comment:
This nested utility class is polluting the Prototype API.
Instead shouldn't this be a general purpose utility class to support generated code for builders ? I.e. what if we need more utility in the future that aren't optional related.

Reply:
Yes, I will create a dedicated `GeneratedBuilder` or similar type.